pkg/phlaredb/bucketindex/index.go

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/cortexproject/cortex/blob/master/pkg/storage/tsdb/bucketindex/index.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Cortex Authors.
+
+package bucketindex
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/oklog/ulid"
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/pyroscope/pkg/phlaredb/block"
+	"github.com/grafana/pyroscope/pkg/phlaredb/sharding"
+	"github.com/grafana/pyroscope/pkg/util"
+)
+
+const (
+	IndexFilename           = "bucket-index.json"
+	IndexCompressedFilename = IndexFilename + ".gz"
+	IndexVersion1           = 1
+	IndexVersion2           = 2 // Added CompactorShardID field.
+	SegmentsFormatUnknown   = ""
+
+	// SegmentsFormat1Based6Digits defined segments numbered with 6 digits numbers in a sequence starting from number 1
+	// eg. (000001, 000002, 000003).
+	SegmentsFormat1Based6Digits = "1b6d"
```

Comment:
Nit: I haven't found any mentions of segments in the package. Should we maybe remove the constants?

Reply:
You're right we should remove all of this.